### PR TITLE
Soft delete line backwards implementation

### PR DESCRIPTION
**Pull request by alunturner**
This work is done to address [issue 456](https://github.com/matrix-org/matrix-rich-text-editor/issues/456).

Implementation has to be done on the web side, not the rust side, as we need to know the width of what is being rendered.

The testing is limited because jsdom does not implement `Selection.modify`. I'm not surprised it doesn't, because the functionality is surprisingly clever and would involve a whole heap of modelling that (I'm guessing) jsdom probably doesn't do at the moment.

### CODE REVIEW

File: platforms/web/lib/composer.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ export function processInput(
             return action(composerModel.backspace(), 'backspace');
         case 'deleteWordBackward':
             return action(composerModel.backspace_word(), 'backspace_word');
+        case 'deleteSoftLineBackward': {
+            document
+                .getSelection()
+                ?.modify('extend', 'backward', 'lineboundary');
+            document.dispatchEvent(new CustomEvent('selectionchange'));
```
Comment (florianduros) [inline platforms/web/lib/composer.ts:78]: If we can't call modify (ie no selection), I think we shouldn't dispatch the `selectionchange` event

**Review by jonnyandrew — APPROVED** [review]
First I heard about this command! Looks good to me

**Review by florianduros — CHANGES_REQUESTED** [review]
Looks good, thanks!

A small improvement can be made

File: platforms/web/lib/composer.ts (hunk 1 above)
Comment (alunturner) [reply platforms/web/lib/composer.ts:78]: That's a fair point, I'll do that to appease TS, but the case that returns a null selection is pretty niche

Comment (alunturner) [reply platforms/web/lib/composer.ts]: Changes applied

**Review by florianduros — APPROVED** [review]

**Pull request merged**